Extract Go snippets

erdemyerebasmaz opened the pull request:
This PR adds Go snippets to
- #72 

##### TODO:
- [ ] Add CI for Go snippets

JssDWt commented on snippets/go/go.sum:
```diff
@@ -0,0 +1,2 @@
+github.com/breez/breez-sdk-go v0.2.7 h1:9Kemh1y/T2y6y9eIR3ZYYbfYNQYyJ+UfyIoI8xmar90=
```
let's gitignore this file to make it easier to switch versions

JssDWt commented on snippets/go/main.go:
```diff
@@ -0,0 +1,13 @@
+package main
```
There is a binary file called `main` committed

JssDWt commented on snippets/go/send_payment.go:
```diff
@@ -0,0 +1,23 @@
+	bolt11 := "bolt11 invoice"
+	// The `amountMsat` param is optional and should only passed if the bolt11 doesn't specify an amount.
+	// The amountMsat is required in case an amount is not specified in the bolt11 invoice'.
+	amountMsat := uint64(3000000)
```
```suggestion
	amountMsat := uint64(3_000_000)
```

JssDWt commented on snippets/go/send_onchain.go:
```diff
@@ -0,0 +1,55 @@
+func StartReverseSwap() {
+	// ANCHOR: start-reverse-swap
+	destinationAddress := "bc1.."
+	sendAmountSat := uint64(50000)
```
```suggestion
	sendAmountSat := uint64(50_000)
```

JssDWt commented on snippets/go/send_onchain.go:
```diff
@@ -0,0 +1,55 @@
+
+func GetCurrentFees() {
+	// ANCHOR: estimate-current-reverse-swap-total-fees
+	sendAmountSat := uint64(50000)
```
```suggestion
	sendAmountSat := uint64(50_000)
```

JssDWt commented on snippets/go/receive_onchain.go:
```diff
@@ -0,0 +1,59 @@
+
+func GetChannelOpeningFees() {
+	// ANCHOR: get-channel-opening-fees
+	amountMsat := uint64(10000)
```
this can be passed in as a parameter. Or make it 3_000_000.

JssDWt reviewed (commented):
left a few comments.

dangeross reviewed (changes requested):
Looks good, just a small anchor fix needed

erdemyerebasmaz commented:
Addressed feedback.
- Removed & added binary & dependency checksum files to .gitignore
- Added underscore as a separator in number literals
- Corrected get-channel-opening-fees anchor_end
- Farewell to "Hello World"

JssDWt reviewed (approved):
LGTM!

dangeross reviewed (approved):
🚀
